### PR TITLE
fix: update mocha chai challenge text, code snippets

### LINKED ISSUE #40332: Flesh-Out Mocha/Chai Functional Test
Following on from: https://github.com/freeCodeCamp/freeCodeCamp/pull/40330 (thanks, @nhcarrigan )

The tests for this lesson/challenge need to be expanded (more included): 
https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/06-quality-assurance/quality-assurance-and-testing-with-chai/simulate-actions-using-a-headless-browser.md

_Note: Only once the mentioned PR is merged, will this make sense_

This will likely involve:
1) Including an endpoint on the boilerplate `server.js` to fetch/return the file contents
2) Testing for the existence of the relevant code, after hitting the endpoint (Regex)

Otherwise, this issue is open to discuss a better alternative (non-regex).

### PR DESCRIPTION
**Pull request by scissorsneedfoodtoo**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Partially addresses #38837
Closes #40332

This PR updates some of the text and code snippets for the Quality Assurance and Testing with Chai challenges based on updates in https://github.com/freeCodeCamp/boilerplate-mochachai/pull/42.

These updates should be deployed before https://github.com/freeCodeCamp/boilerplate-mochachai/pull/42 is merged.

Also, here's the completed project on Replit for testing: https://replit.com/@scissorsneedfoo/boilerplate-mochachai

### CODE REVIEW

**Review by naomi-lgbt — DISMISSED** [review]
This is looking good - should help make things a bit clearer for these challenges.

**Review by moT01 — COMMENTED** [review]
I would be fine adding this as-is since it's quite an improvement.

But some things I noticed...
the two "run functional tests on an api response" lessons ask for users to check certain things that they can only find from the test text. For instance, they ask for to check the `status`, `type`, `body.name`, and `body.surname`, but the only place to know what some of those values should be is by looking at the test text.

The `simulate-actions-using-a-headless-browser` lesson doesn't really have any tests. If users add the wrong code, there's no way for them to really know. I was thinking a way to test that lesson might be to add a test for it. Something like this right after the setup:
  ```
  suite('Headless browser exists', function () {
    test('headless browser exists', function() {
      assert.exists(browser);
    });
  });
  ```
That would go in the boilerplate - then we would have to adjust the test for it to check `data.slice(0, 5)` instead of `data.slice(0, 4)`. And the last few lessons after it would have to add one to the slice.

I worked on this locally, the suggestion in the `simulate` lesson says to use:
```
Browser.localhost('example.com', process.env.PORT || 3000);
```
But that wasn't working for me - I just used the first method:
```
Browser.site = 'localhost:3000'; // Your URL here
```
So I think we could probably remove that example.

These probably aren't part of the issue, but we could maybe fix up these few things. I can make some commits to do that if you want @scissorsneedfoodtoo

**Comment by moT01** [comment]
Alright, I made a few commits to address the issues I found. Here's an updated project that passed https://boilerplate-mochachai-1.mot01.repl.co

**Review by moT01 — APPROVED** [review]
I tested it quite thoroughly, LGTM 🎉

File: curriculum/challenges/english/06-quality-assurance/quality-assurance-and-testing-with-chai/simulate-actions-using-a-headless-browser.md [hunk 1]
```diff
@@ -9,37 +9,41 @@ dashedName: simulate-actions-using-a-headless-browser
 
 As a reminder, this project is being built upon the following starter project on [Replit](https://replit.com/github/freeCodeCamp/boilerplate-mochachai), or cloned from [GitHub](https://github.com/freeCodeCamp/boilerplate-mochachai/).
 
-In the next challenges we are going to simulate the human interaction with a page using a device called 'Headless Browser'.
+In the next challenges, you'll simulate human interaction with a page by using a headless browser.
 
-A headless browser is a web browser without a graphical user interface. This kind of tool is particularly useful for testing web pages, as it is able to render and understand HTML, CSS, and JavaScript the same way a browser would.
+Headless browsers are web browsers without a GUI. Headless browsers are particularly useful for testing web pages, as they're able to render and interpret HTML, CSS, and JavaScript the same way a regular browser would.
```
[the remark this answers is not in the document]
Comment (scissorsneedfoodtoo) [reply curriculum/challenges/english/06-quality-assurance/quality-assurance-and-testing-with-chai/simulate-actions-using-a-headless-browser.md]: Awesome, thanks for catching that. This updated text LGTM.

**Comment by scissorsneedfoodtoo** [comment]
Thanks for your thorough review and fixes both here and in the boilerplate, @moT01. They really improve the clarity and quality of these challenges :+1: :+1:

**Review by jatin837 — APPROVED** [review]

**Review by naomi-lgbt — APPROVED** [review]
Ran through these challenges with the new boilerplate - everything seems clear and worked fine locally. Nice work Kris!

**Pull request merged**